python/foglamp/services/core/api/filters.py

```diff
@@ -0,0 +1,245 @@
+
+        # Fetch the new created filter: get category items
+        category_info = await cf_mgr.get_category_all_items(category_name = filter_name)
+        if category_info is None:
+            message = "No such '%s' filter found" % filter_name
+            raise Exception(message)
+
+    except Exception as ex:
+        _LOGGER.error("Add filter, caught exception: " + str(ex))
+        raise web.HTTPInternalServerError(reason=str(ex))
+
+    # Success: return new filter content
+    return web.json_response({'filter': filter_name,
+                              'description': filter_desc,
+                              'value': category_info})
+
+"""
+    Add filter names to "filter" item in {service_name}
+
+    PUT /foglamp/filter/{service_name}/pipeline
+ 
+    'pipeline' is the array of filter category names to set
+    into 'filter' default/value properties
+
+    :Example: set 'pipeline' for service 'NorthReadings_to_PI'
+    curl -X PUT http://localhost:8081/foglamp/filter/NorthReadings_to_PI/pipeline -d 
+    '{
+        "pipeline": ["Scale10Filter", "Python_assetCodeFilter"],
+    }'
+
+    Configuration item 'filter' is added to {service_name}
+    or updated with the pipeline arryay
+
+    Returns the filter pipeline on success:
+    {"pipeline": ["Scale10Filter", "Python_assetCodeFilter"]} 
+
+    Query string parameters:
+    - append_filter=true|false       Default true
+    - allow_duplicates=true|false    Default true
+
+    NOTE: the method also adds the filters category names under
+    parent category {service_name}
+"""
+
+async def add_filters_pipeline(request):
+    try:
+        # Get inout data
+        data = await request.json()
+        # Get filters list
+        filter_list = data.get('pipeline')
+        # Get filter name
+        service_name = request.match_info.get('service_name', None)
+        # Item name to add/update
+        config_item = "filter"
+
+        # Get configuration manager instance
+        cf_mgr = ConfigurationManager(connect.get_storage_async())
+        # Fetch the filter items: get category items
+        category_info = await cf_mgr.get_category_all_items(category_name = service_name)
+        if category_info is None:
+            # Error filter_name doesn't exist
+            message = "No such '%s' category found." % service_name
+            return web.HTTPNotFound(reason=message)
+
+        # Check whether config_item already exists
+        if (config_item in category_info):
+            # We just need to update the value of config_item
+            # with the "pipeline" property
+            # Check whether we want to replace or update the list
+            # or we allow duplicate entries in the list
+            # Default: append and allow duplicates
+            append_filter = 'true'
+            allow_duplicates = 'true'
+            if 'append_filter' in request.query and request.query['append_filter'] != '':
```

Comment:
@MarkRiddoch I believe`append_filter and allow_duplicates` keys should be wrapped inside `data` if they are given and should not be the part of request query as it is PUT
Like below
`curl -X PUT http://localhost:8081/foglamp/filter/anyservice/pipeline -d '{"pipeline": ["filter1","filter2"], "append_filter": "true|false", "allow_duplicate": "true|false"}'`

Reply:
@ashish-jabble I don't think so. It's shouldn't be part of the payload, it's just a modifier.

There was a wrong similar one with keep_original_items in the payload, moved  some weeks ago into a query string parameter (it was for POST /foglamp/category

Reply:
done

Reply:
fixed